### ISSUE TITLE
obs-ndi.so library installs in the wrong directory of Ubuntu 19.10 (64 bit)

### ISSUE THREAD

**Issue by computer-wizard** [Palakis/obs-ndi#461]
Version 4.9.0 deb packages.
The obs-ndi.so library is installed in the /usr/lib/x86_64-linux-gnu/obs-plugins/ directory.  This has the effect that NDI doesn't show up in OBS (version 25.0.8).  This will be solved by making a softlink to
this library in the /usr/lib/obs-plugins/ directory.

Note: Obs installed from ppa or obsproject.com

Regards,

Boudewijn (Bob) Tenty

**Comment by tt2468** [comment]
This appears to be an issue with any recent Ubuntu edition.

**Comment by enesdemirag** [comment]
Yes. I also cannot see NDI plugin inside OBS on my Ubuntu 20.04.
- OBS installed from ppa
- NDI installed from .deb files

**Comment by favrik** [comment]
It is happening on Ubuntu 16.04 LTS,   I did the soft link, but it did not work.  I also copied the library file to the directory, but didn't work.

**Comment by freakinbox** [comment]
> It is happening on Ubuntu 16.04 LTS, I did the soft link, but it did not work. I also copied the library file to the directory, but didn't work.

    use    
    `sudo ln -s /usr/lib/x86_64-linux-gnu/obs-plugins/obs-ndi.so /usr/lib/obs-plugins/`

Took me a bit to figure out that I had to link directly to the plugin itself and not just the directory. Worked perfectly fine after I did that.

**Comment by favrik** [comment]
thanks @freakinbox !    I opened OBS via terminal, and was able to see a different error (Qt_5 not found),  so my issue is different now, but thanks!!

**Comment by jcward** [comment]
Same here, Ubuntu 18.04, OBS 25.0.8 from ppa, obs-ndi 4.9.0. Thanks for the note!

To expound on the OP (and note to my future self ;)

**How can I tell if this is my problem?**

Assuming you've downloaded and installed the .deb files (`sudo dpkg -i xyz.deb`) from the [releases](https://github.com/Palakis/obs-ndi/releases) page ([libndi4_4.5.1-1_amd64.deb](https://github.com/Palakis/obs-ndi/releases/download/4.9.0/libndi4_4.5.1-1_amd64.deb) and don't miss the [obs-ndi-4.9.0-1_amd64.deb](https://github.com/Palakis/obs-ndi/releases/download/4.9.0/obs-ndi_4.9.0-1_amd64.deb) down lower in the list.)

Now, let's find out where `obs-ndi.so` got installed. First do a `sudo updatedb`, then `locate obs-ndi.so` shows it's installed at `/usr/lib/x86_64-linux-gnu/obs-plugins/obs-ndi.so`

So then do a `ls -la /usr/lib/x86_64-linux-gnu/obs-plugins/` and see if there are no other `.so` files in that directory, and `ls -la /usr/lib/obs-plugins` to see if there are a bunch of .so files there.

If so, then this is your problem.

**How to fix it:**

Simply symlink the .so file into the directory with the other plugins (and restart obs afterward):

```
cd /usr/lib/obs-plugins
sudo ln -s /usr/lib/x86_64-linux-gnu/obs-plugins/obs-ndi.so .
```

Now it's working!
![image](https://user-images.githubusercontent.com/2192439/81235780-f8d9d000-8fb8-11ea-8355-8f7ce7f4fcd1.png)

**Comment by ajmiley** [comment]
You have just made my day, morning, week, month year!!! I finally came across a method that worked. I hope and pray that I can pay back someone in the future who needs help as I did with this so called simple method. The solution here above is simple, but it took me so long to find it. I kept getting shut down because I did not have "permission" to do a copy and paste inside Ubuntu 18.04. These two commands above solved my challenge of getting OBS to recognize NDI under tools menu. What a gift. Thank you!!! I am looking forward to mastering Ubuntu, but not sure if I will actually get into the art of programming.
